doc/manual/source/man/dnst-signzone.rst

```diff
@@ -0,0 +1,108 @@
+dnst-signzone
+===============
+
+Synopsis
+--------
+
+:program:`dnst signzone` [``OPTIONS``] ``<ZONEFILE>`` ``<KEY>...``
+
+Description
+-----------
+
+**dnst signzone** signs the zone with the given key(s).
+
+Keys must be specified by their base name (usually ``K<name>+<alg>+<id>``),
+i.e. WITHOUT the ``.private`` or ``.key`` extension. Both ``.private`` and
+``.key`` files are required.
+
+A date can be a timestamp (seconds since the epoch), or of the form
+<YYYYMMdd[hhmmss]>.
+
+
+Options
+-------
+
+.. option:: <ZONEFILE>
+
+      The zonefile to sign.
+
+.. option:: <KEY>...
+
+      The keys to sign the zone with.
+
+.. option:: -b
+
+      Use a more readable layout in the signed zone file and print comments on
+      DNSSEC records.
+
+.. option:: -d
+
+      Do not add used keys to the resulting zone file.
+
+.. option:: -e <DATE>
+
+      Set the expiration date. Defaults to 4 weeks from now.
+
+.. option:: -f <FILE>
+
+      Write signed zone to file. Use ``-f -`` to output to stdout. Defaults to
+      ``<ZONEFILE>.signed``.
+
+.. option:: -i <DATE>
+
+      Set the inception date. Defaults to now.
+
+.. option:: -o <domain>
+
+      Set the origin for the zone (for zonefiles with relative names and no
```

Comment:
This is correct but perhaps it would be better to say that this is only needed for zones that both contain relative labels AND lack an `$ORIGIN` directive.

Reply:
Yes. It needs to be an absolute domain name. Also, signzone or rather domain's Name (::from_str?) turns any non-dot-terminated domain into an absolute/dot-terminated domain. So `dnst signzone -o origin` and `dnst signzone -o origin.` do the same.